pkg/api/customization/clusterregistrationtokens/formatter.go

```diff
@@ -29,16 +29,27 @@ func Formatter(request *types.APIContext, resource *types.RawResource) {
 		url := getURL(request, token)
 		resource.Values["insecureCommand"] = fmt.Sprintf(insecureCommandFormat, url)
 		resource.Values["command"] = fmt.Sprintf(commandFormat, url)
+		resource.Values["token"] = token
+		resource.Values["manifestUrl"] = url
+
+		rootURL := getRootURL(request)
+		agentImageInSetting := settings.AgentImage.Get()
+		agentImage := image.Resolve(agentImageInSetting)
+		// for linux
 		resource.Values["nodeCommand"] = fmt.Sprintf(nodeCommandFormat,
-			image.Resolve(settings.AgentImage.Get()),
-			getRootURL(request),
+			agentImage,
+			rootURL,
 			token,
 			ca)
-		resource.Values["token"] = token
-		resource.Values["manifestUrl"] = url
+		// for windows
+		var agentImageDockerEnv string
+		if agentImage != agentImageInSetting {
+			agentImageDockerEnv = fmt.Sprintf("-e AGENT_IMAGE=%s ", agentImage)
```

Comment:
Do we enter this condition when the system default registry is set?  Can we do the following which would be more intuitive:
```
systemDefaultRegistry:=settings.SystemDefaultRegistry.Get()
if systemDefaultRegistry != "" {
    ...
}
```

Reply:
Updated